astropy/wcs/src/util.c

```diff
@@ -26,12 +26,21 @@ void set_invalid_to_nan(
     #define NAN (INF-INF)
   #endif
 
+  // Note that stat is a bit mask, so we need to mask only some of
+  // the coordinates depending on the bit mask values.
+
   n = NAN;
 
   for ( ; s != s_end; ++s) {
     if (*s) {
+      int bit = 1;
       for (i = 0; i < nelem; ++i) {
-        *d++ = n;
+        if (*s & bit) *d = n;
```

Comment:
I'm not in favor of dropping braces (even if allowed) and PEP7 also says "braces are required everywhere, even where C permits them to be omitted". But this is pure nitpicking and probably very subjective as well. Feel free to leave this as is if you've a different opinion.

Reply:
Fixed!